configure.ac

```diff
@@ -5361,7 +5361,7 @@ AC_CACHE_CHECK([for sockaddr_alg], [ac_cv_struct_sockaddr_alg],
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #		include <sys/types.h>
 #		include <sys/socket.h>
-#		include <linux/if_alg.h>]], [[struct sockaddr_alg s]])],
+@%:@		include <linux/if_alg.h>]], [[struct sockaddr_alg s]])],
```

Comment:
It looks like that we need to use different syntax for each case.
I am not sure it will be acceptable to other people..
I think that using the same syntax for C header is more proper than not. WDYT?

Reply:
We already use quadrigraphs for writing brackets, so there's precedent for this:

https://github.com/python/cpython/blob/601ae09f0c8eda213b9050892f5ce9b91f0aa522/configure.ac#L7024-L7026

https://github.com/python/cpython/blob/601ae09f0c8eda213b9050892f5ce9b91f0aa522/configure.ac#L413-L415

https://github.com/python/cpython/blob/601ae09f0c8eda213b9050892f5ce9b91f0aa522/configure.ac#L659

https://github.com/python/cpython/blob/601ae09f0c8eda213b9050892f5ce9b91f0aa522/configure.ac#L671

etc.

Use of quadrigraphs also aligns with the [Autoconf recommendations](https://www.gnu.org/software/autoconf/manual/autoconf-2.71/html_node/Coding-Style.html):

> Feel free to use various tricks to prevent auxiliary tools, such as syntax-highlighting editors, from behaving improperly [...]
> [...] For the same reasons, avoid:
> `test $[#] != 0`
> and use:
> `test $[@%:@] != 0`

I would expect anyone hacking `configure.ac` to acquire the needed Autoconf skills to do so.

BTW; this PR is deliberately designed to not introduce changes in the generated `configure` script.

Reply:
> It looks like that we need to use different syntax for each case.

Yes, of course.